src/coreComponents/mesh/CellElementSubRegion.hpp

```diff
@@ -16,24 +16,50 @@
 #ifndef GEOSX_MESH_CELLELEMENTSUBREGION_HPP_
 #define GEOSX_MESH_CELLELEMENTSUBREGION_HPP_
 
-#include "CellBlock.hpp"
+#include "mesh/generators/CellBlockABC.hpp"
+#include "mesh/NodeManager.hpp"
+#include "mesh/FaceManager.hpp"
+#include "mesh/utilities/ComputationalGeometry.hpp"
+#include "ElementSubRegionBase.hpp"
+
 
 namespace geosx
 {
 
+class MeshLevel;
+
 /**
  * @class CellElementSubRegion
  * Class deriving from CellBlock further specializing the element subregion
  * for a cell element. This is the class used in the physics solvers to
  * represent a collection of mesh cell elements
  */
-class CellElementSubRegion : public CellBlock
+class CellElementSubRegion : public ElementSubRegionBase
 {
 public:
 
+  /// Alias for the type of the element-to-node map
+  using NodeMapType = InterObjectRelation< array2d< localIndex, cells::NODE_MAP_PERMUTATION > >;
+  /// Alias for the type of the element-to-edge map
+  using EdgeMapType = FixedOneToManyRelation;
+  /// Alias for the type of the element-to-face map
+  using FaceMapType = FixedOneToManyRelation;
   /// Type of map between cell blocks and embedded elements
   using EmbSurfMapType = InterObjectRelation< ArrayOfArrays< localIndex > >;
 
+  /**
+   * @brief Const getter for the catalog name.
+   * @return the name of this type in the catalog
+   */
+  static const string catalogName()
+  { return "CellElementSubRegion"; } // FIXME I don't know what I am doing
```

Comment:
Maybe remove the comment, because this return value looks fine?

Reply:
done